File: ecotrack-frontend/src/services/notificationService.js
import api from './api';

const notificationService = {
  getMyNotifications: (params) => api.get('/notifications/my', { params }).then((r) => r.data.data),
  getUnreadCount: () => api.get('/notifications/unread-count').then((r) => r.data.data),
  markAsRead: (id) => api.patch(`/notifications/${id}/read`).then((r) => r.data.data),
  markAllAsRead: () => api.patch('/notifications/read-all').then((r) => r.data.data),
  getAuditLogs: (params) => api.get('/notifications/audit-logs', { params }).then((r) => r.data.data),
};

export default notificationService;

File: ecotrack-backend/auth-service/src/main/java/com/ecotrack/notification/service/NotificationService.java
package com.ecotrack.notification.service;

import com.ecotrack.notification.dto.response.AuditLogResponse;
import com.ecotrack.notification.dto.response.NotificationResponse;
import com.ecotrack.notification.enums.NotificationType;
import com.ecotrack.notification.enums.RoleType;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface NotificationService {

    /** Creates a notification for one user. Called by the RabbitMQ consumers. */
    void notify(Long userId, RoleType userRole, String title, String message, NotificationType type);

    Page<NotificationResponse> getMyNotifications(Long userId, Pageable pageable);

    long getUnreadCount(Long userId);

    void markAsRead(Long notificationId, Long userId);

    void markAllAsRead(Long userId);

    /** Simple audit trail, populated by the same consumers. */
    void logAudit(String action, String details);

    Page<AuditLogResponse> getAuditLogs(Pageable pageable);

    Page<AuditLogResponse> getFilteredAuditLogs(String role, String region, Integer status, String search, Pageable pageable);
}
